### ISSUE TITLE
If Serialisation null bug

### ISSUE THREAD

**Issue by GCHQDev404** [gchq/koryphe#302]
Reported bug.

"Serialise and deserialise the `IF` function with a null condition it sets the condition to false, meaning you can't use the predicate"

**Comment by t92549** [comment]
I don't think so, the default value should be null: https://github.com/gchq/koryphe/blob/812009bd172b19705923c86c6bed5baa34ec13f8/core/src/main/java/uk/gov/gchq/koryphe/impl/function/If.java#L63
